contracts/secret-storage/SecretStore.sol

```diff
@@ -447,6 +447,23 @@ contract SecretStore is
         delete secretStores[_enclaveAddress].ackSecretIds;
     }
 
+    function _renounceSecretsPreUpdate(
+        address _enclaveAddress,
+        uint256 _markAliveTimeout,
+        address _recipient
+    ) internal {
+        _slashStore(_enclaveAddress, block.timestamp, _markAliveTimeout, _recipient);
+        secretStores[_enclaveAddress].lastAliveTimestamp = block.timestamp;
+    }
+
+    function _renounceSecretsPostUpdate(
+        address _enclaveAddress,
+        uint256 _storageOccupied
+    ) internal {
+        secretStores[_enclaveAddress].storageOccupied -= _storageOccupied;
```

Comment:
we have to make sure storageOccupied get to the value zero before deregistration and remove stake. i.e. no secrets assigned.

Reply:
already being checked in the respective functions